dolphin-dnn/src/main/java/edu/snu/dolphin/dnn/layers/LayerBase.java

```diff
@@ -39,13 +46,18 @@ public final int getIndex() {
   }
 
   /**
-   * @return the number of layer output nodes.
+   * @return the shape of inputs.
    */
-  public final int getNumOutput() {
-    return this.numOutput;
+  public final int[] getInputShape() {
```

Comment:
The only reason for having this public method and the one below seems to be to transform labels to one-hot vectors. Other than that, these methods look a bit redundant. Besides, the label-vector transformation can be done without the output shape known beforehand, by looking at the output themselves just before back propagation starts (around `NeuralNetwork.backPropagateTo`?). Am I missing something?